source/common/common/hash.cc

```diff
@@ -34,4 +34,40 @@ uint64_t MurmurHash::murmurHash2_64(absl::string_view key, uint64_t seed) {
   return hash;
 }
 
+CharStarSet::~CharStarSet() {
+  std::vector<char*> keys;
+  keys.reserve(hash_set_.size());
+  for (char* p : hash_set_) {
+    keys.push_back(p);
+  }
+  hash_set_.clear();
+  for (char* p : keys) {
+    delete[] p;
```

Comment:
For my own education, what's going on here? Is it not valid to delete these in-place in the hashset?